### PR TITLE
use v1 CSIDriver object for 1.18+ clusters; clean up pre-1.17 versions

### PR DESCRIPTION
**Pull request by mattcary**
/kind failing-test

**What this PR does / why we need it**:
CI tests are failing due to trying to install v1beta1 versions of CSIDriver objects on 1.22 clusters. This PR changes deployments to use the v1 CSIDriver object on all 1.18+ clusters.

```release-note
v1 CSIDriver resources are deployed for 1.18+ clusters.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,19 +33,14 @@ This plugin is compatible with CSI versions [v1.2.0](https://github.com/containe
...
-| v0.3.x (beta)                        | yes  | yes   | yes   |
-| v0.4.x (beta)                        | yes  | yes   | yes   |
-| v0.5.x (beta)                        | yes  | yes   | yes   |
-| v0.6.x (beta)                        | yes  | yes   | yes   |
-| v0.7.x (beta)                        | yes  | yes   | yes   |
-| v1.0.x (ga)                          | yes  | yes   | yes   |
-| dev                                  | yes  | yes   | yes   |
+| GCE PD CSI Driver\Kubernetes Version | 1.17+ |
+|--------------------------------------|-------|
+| v0.7.x (beta)                        | yes   |
```
Comment (jingxu97) [inline README.md:38]: so we want to remove the older version information?

Comment (mattcary) [reply README.md:38]: yes, it's just noise I think, especially since I also removed the older overlays as well.

**Pull request merged**